### PR TITLE
feat: add v13 modals guide (#1060)

### PR DESCRIPTION
**Pull request by lolollllo**
**Please describe the changes this PR makes and why it should be merged:**
Adds a guide on how to use modals for discord.js v13

Credits to @suneettipirneni for creating PR #1060 which provided most of the code.

### CODE REVIEW

File: guide/interactions/modals.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+
+Restart your bot and invoke the `/ping` command again. You should see a popup form resembling the image below:
+
+<img width=450 src="./images/modal-example.png">
+
+## Receiving modal submissions
+
+Modals are received via an interaction. You can check if a given interaction is a modal by invoking the `#isModalSubmit()` method:
```
Comment (Jiralite) [inline guide/interactions/modals.md:89]:
```suggestion
Modals are received via an interaction. You can check if a given interaction is a modal by invoking the `Interaction#isModalSubmit()` method:
```

File: guide/interactions/modals.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+```js {2}
+client.on('interactionCreate', interaction => {
+	if (!interaction.isModalSubmit()) return;
+	console.log(interaction);
+});
+```
+
+## Extracting data from modal submissions
+
+You'll most likely need to read the data sent by the user in the modal. You can do this by accessing the `#fields` instance field on the interaction. From there you can call `#getTextInputValue` with the custom id of the text input to get the value.
```
Comment (Jiralite) [inline guide/interactions/modals.md:100]:
```suggestion
You'll most likely need to read the data sent by the user in the modal. You can do this by accessing the `fields` property from the interaction. From there, you can call `getTextInputValue()` with the custom id of the text input to get the value.
```

File: guide/interactions/modals.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+You are still missing one of these steps - adding inputs. Adding inputs is similar to adding components to messages.
+
+::: warning
+If you're using typescript you'll need to specify the type of components your action row holds. This can be done by specifying the generic parameter in `MessageActionRow`.
+
+```diff
+- new MessageActionRow()
++ new MessageActionRow<ModalActionRowComponent>()
+```
+:::
```
[the remark this answers is not in the document]
Comment (almostSouji) [reply guide/interactions/modals.md:46]: i think making the guide typescript compatible is a larger endeavour and not something we should selectively apply to a single case

[the remark this answers is not in the document]
Comment (almostSouji) [reply guide/interactions/modals.md:47]: It makes sense to cover it one way or another, this is fine for now, IMO

**Review by almostSouji — APPROVED** [review]

**Pull request merged**